src/main/java/uk/gov/companieshouse/documentsigningapi/coversheet/CoverSheetService.java

```diff
@@ -1,39 +1,188 @@
+import uk.gov.companieshouse.documentsigningapi.dto.CoverSheetDataDTO;
 import uk.gov.companieshouse.documentsigningapi.exception.CoverSheetException;
 import uk.gov.companieshouse.documentsigningapi.logging.LoggingUtils;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 
+import static org.apache.commons.lang.StringUtils.isEmpty;
 import static org.apache.pdfbox.pdmodel.common.PDRectangle.A4;
 
 @Service
 public class CoverSheetService {
 
+    private static class Link {
+        private final String text;
+        private final String url;
+
+        private Link(String text, String url) {
+            this.text = text;
+            this.url = url;
+        }
+    }
+
+    private static class Font {
+        private final PDFont pdFont;
+        private final float size;
+
+        private Font(PDFont pdFont, float size) {
+            this.pdFont = pdFont;
+            this.size = size;
+        }
+    }
+
+    private static class Position {
+        private final float x;
+        private final float y;
+
+        private Position(float x, float y) {
+            this.x = x;
+            this.y = y;
+        }
+    }
+
     private final LoggingUtils logger;
 
-    public CoverSheetService(LoggingUtils logger) {
+    private final String imagesPath;
+
+    // Coversheet measurements
+    private static final float DEFAULT_LEFT_MARGIN = 25;
+    private static final float INFORMATION_SECTION_IMAGE_HEIGHT = 25;
+    private static final float INFORMATION_SECTION_IMAGE_WIDTH = 25;
+    private static final float OFFSET_TO_RIGHT_OF_IMAGES = 70;
+
+    // Coversheet Strings
+    private static final String CERTIFIED_DOCUMENT_TYPE = "Certified document";
+    private static final String DOCUMENT_SIGNED_TEXT = "This file has been electronically signed by Companies House.";
+    private static final String EMAIL_HELPTEXT = "This file can be emailed. You cannot make changes to this file.";
+    private static final String PAGE_HEADING = "Companies House";
+    private static final String PAGE_SPACER = "___________________________________________________";
+    private static final String PRINTER_HELPTEXT = "This file is only valid in digital form as it contains a unique " +
+        "electronic signature. Printed copies of this file will not be accepted.";
+
+    private static final String SIGNATURE_HELPTEXT_LINE_1 = "This file has an embedded electronic signature " +
+            "that is only accessible in Adobe";
+    private static final String SIGNATURE_HELPTEXT_LINE_2_START = "Reader. You can ";
+    private static final String SIGNATURE_HELPTEXT_LINE_2_LINK = "download Adobe Reader";
+    private static final String SIGNATURE_HELPTEXT_LINE_2_END = " for free.";
+    private static final String ADOBE_DOWNLOAD_URL = "https://get.adobe.com/reader/";
+
+    private static final String VIEW_FILE_HEADING = "View this file and signature in Adobe Reader";
```

Comment:
Do you think there would be any benefit to possibly extracting this to be drawn from environment variables to allow quick modification of cover sheet text without needing actual code changes?

Reply:
It would probably be quite a few environment variables! Food for thought though, depending on how often it turns out we need to update the coversheet content?